stor/tests/test_integration_dx.py

```diff
@@ -0,0 +1,320 @@
+import gzip
+import os
+import pytest
+import time
+import unittest
+from unittest import skipIf
+
+import dxpy
+import mock
+import six
+
+from stor import exceptions
+from stor import NamedTemporaryDirectory
+from stor import Path
+from stor.dx import DXPath
+from stor.test import DXTestCase
+from stor.tests.shared import assert_same_data
+from stor.tests.test_integration import BaseIntegrationTest
+import stor
+import stor.tests.test_integration as test_integration
+
+
+class DXIntegrationTest(BaseIntegrationTest.BaseTestCases, DXTestCase):
+    def setUp(self):
+        self.vcr_enabled = False  # don't want vcr playback for integration tests
+        super(DXIntegrationTest, self).setUp()
+        if not os.environ.get('DX_AUTH_TOKEN'):
+            raise unittest.SkipTest(
+                'DX_AUTH_TOKEN env var not set. Skipping integration test')
+
+        self.setup_temporary_project()
+        self.project = DXPath('dx://' + self.project + ':')
+        self.test_dir = self.project / 'test'
+
+    def tearDown(self):
+        super(DXIntegrationTest, self).tearDown()
+
+    def test_login_auth(self):
+        def mock_header(r, security_context):
+            auth_header = security_context["auth_token_type"] + " " + \
+                          security_context["auth_token"]
+            r.headers[b'Authorization'] = auth_header.encode()
+            return r
+        # dxpy.AUTH_HELPER gets set upon login and called with each api which we mock out here
+        with mock.patch('dxpy.AUTH_HELPER', autospec=True) as mock_auth:
```

Comment:
integration tests should not use mocks, not sure why this is here . (i could see using this in test_dx.py tho)

Reply:
I agree. I was thrown off by login authentication stuff for swift `test_cached_auth_and_auth_invalidation` being in `test_integration_swift.py` which also uses mocks (so does another test case for swift :) ). Putting this in unit test for DX though.